pkg/git/git.go

```diff
@@ -3,37 +3,37 @@ package git
 import (
 	"context"
 	"fmt"
-	"github.com/kris-nova/logger"
-	"github.com/weaveworks/eksctl/pkg/git/executor"
 	"io/ioutil"
 	"os"
 	"os/exec"
 	"time"
+
+	"github.com/kris-nova/logger"
+	"github.com/weaveworks/eksctl/pkg/git/executor"
 )
 
+// Cloner can clone git repositories
+type Cloner interface {
+	CloneRepo(cloneDirPrefix string, branch string, gitURL string) (string, error)
+}
+
 // Client can perform git operations on the given directory
 type Client struct {
 	executor executor.Executor
 	dir      string
-	user     string
-	email    string
 }
 
 // NewGitClient returns a client that can perform git operations
-func NewGitClient(ctx context.Context, user string, email string, timeout time.Duration) *Client {
+func NewGitClient(ctx context.Context, timeout time.Duration) *Client {
```

Comment:
Unrelated to this PR, but contexts shouldn't be stored, but passed to the operations.